### PR TITLE
Added Prow go-vet job description

### PR DESCRIPTION
**Pull request by rabugopsl**
Added go-vet test for openshift/installer

### CODE REVIEW

File: cluster/ci/config/prow/config.yaml [hunk 1]
```diff
@@ -2380,6 +2380,26 @@ presubmits:
         args:
           - ./hack/tf-lint.sh
 
+  - name: go-vet
```
Comment (alejovicu) [inline cluster/ci/config/prow/config.yaml]: nit -- this is usually ci-pull-$org-$repo-$name

Comment (rabugopsl) [reply cluster/ci/config/prow/config.yaml]: Done

File: cluster/ci/config/prow/config.yaml [hunk 2]
```diff
@@ -2380,6 +2380,26 @@ presubmits:
...
+      containers:
+      - image: quay.io/coreos/golang-testing
+        env:
+        - name: IS_CONTAINER
+          value: 'TRUE'
+        command:
+        - sh
+        args:
+        - ./hack/go-vet.sh
+        - "./installer/..."
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/config.yaml]: You should be able to do `./...`, no?

Comment (rabugopsl) [reply cluster/ci/config/prow/config.yaml]: You're right. The job has been updated

File: cluster/ci/config/prow/config.yaml [hunk 3]
```diff
@@ -2380,7 +2380,50 @@ presubmits:
...
+      - image: quay.io/coreos/shellcheck-alpine:v0.5.0
+        env:
+        - name: IS_CONTAINER
+          value: "TRUE"
+        command:
+        - sh
+        args:
+        - ./hack/shellcheck.sh
+
+  - name: ci-pull-openshift-installer-go-vet-batch1
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/config.yaml]: I strongly do not suggest doing this, just keep one `go vet` incantation.

Comment (rabugopsl) [reply cluster/ci/config/prow/config.yaml]: The job has been updated accordingly

**Pull request closed without merging**